README.md

```diff
@@ -59,23 +59,58 @@ The library comprises two packages:
-        )
-    ```
-
-    Adding this removes the need for `@import` statements in your Twirl templates.
-    If you prefer not to use this mechanism, import the components in any template that uses them as
-    follows:
-
-    ```scala
-    @import uk.gov.hmrc.hmrcfrontend.views.html.components._
-    ```
+1. Then to use our components and helpers in your templates, you will need to import the component or helper from its
+   package `uk.gov.hmrc.(govuk|hmrc)frontend.views.html.(components|helpers)`.
+
+   For most components the parameters that can be provided are encapsulated in a viewmodel, comprised of case classes
+   that live within a subpackage of `uk.gov.hmrc(govuk|hmrc)frontend.views.viewmodels` and are aliased for use
+   under `uk.gov.hmrc(govuk|hmrc)frontend.views.html.components`.
+
+   ```scala
+   /*
+   Import all components, viewmodels, and their implicits. Most succinct import method, but may require additional 
+   imports to resolve ambiguous import compilation errors for some viewmodels. Will also cause unused import warnings.
+   */
+   @import uk.gov.hmrc.hmrcfrontend.views.html.components._                  
+
+   /*
+   Import specific components and viewmodels, and all implicits. Avoids possibility of ambiguous import compilation 
+   errors and unused import warnings.
+   */
+   @import uk.gov.hmrc.hmrcfrontend.views.html.components.HmrcTimeline      /* component */
+   @import uk.gov.hmrc.hmrcfrontend.views.html.components.{Timeline, Event} /* viewmodel case classes */
+   @import uk.gov.hmrc.hmrcfrontend.views.html.components.implicits._
+
+   @this(hmrcTimeline: HmrcTimeline)
+
+   @()
+
+   @hmrcTimeline(Timeline(
+     events = Seq(
+       Event(
+         title = "Event 1",
+         time = "14 July 2020",
+         datetime = Some("2020-07-15"),
+         content = "Example content"
+       )
+     )
+   ))
+   ```
+
+   > **Notice: Recommended usage pattern changed as of July 2021**
+   >
+   > Previously instead of explicit imports within templates we suggested pulling in all components and helpers (along 
+   > with their implicits) using wildcard imports via `TwirlKeys.templateImports` in `build.sbt`.
+   >
+   > This reduces the boilerplate needed within templates. With the exception being if the class being used caused an 
+   > ambiguous import compilation error, which could occur because a class of the same name was provided by multiple 
+   > imports. This error could be resolved by adding an explicit individual import for the version of the class you 
+   > needed in the template where it was used.
+   >
+   > We no longer recommend the use of the `TwirlKeys.templateImports` configuration. This introduced some complexity, 
+   > and unused imports triggered excessive warning messages which needed to be suppressed or could cause alert fatigue.
+   > However when suppressed there was a possible undesirable side effect of also suppressing meaningful feedback like 
```

Comment:
This is the kind of context we can add into the ADR, I don't know if it's necessary to add into the README.

Reply:
I've kept a note here that recommendation the has changed not to avoid using the twirl keys config but I've trimmed it down. I've added the ADR in now too but seemed like a snippet here for anyone who won't read through our ADRs might help if someone is sure they saw it somewhere but then can't remember where.